google/cloud/aiplatform/training_jobs.py

```diff
@@ -99,14 +103,76 @@ def __init__(
                 aiplatform.init will be used.
             credentials (auth_credentials.Credentials):
                 Optional credentials to use to retrieve the model.
+            training_pipeline_encryption_spec_key_name (Optional[str]):
+                Optional. The Cloud KMS resource identifier of the customer
+                managed encryption key used to protect the training pipeline. Has the
+                form:
+                ``projects/my-project/locations/my-region/keyRings/my-kr/cryptoKeys/my-key``.
+                The key needs to be in the same region as where the compute
+                resource is created.
+
+                If set, this Dataset and all sub-resources of this Dataset will be secured by this key.
+
+                Overrides encryption_spec_key_name set in aiplatform.init.
+            model_encryption_spec_key_name (Optional[str]):
+                Optional. The Cloud KMS resource identifier of the customer
+                managed encryption key used to protect the model. Has the
+                form:
+                ``projects/my-project/locations/my-region/keyRings/my-kr/cryptoKeys/my-key``.
+                The key needs to be in the same region as where the compute
+                resource is created.
+
+                If set, this Dataset and all sub-resources of this Dataset will be secured by this key.
+
+                Overrides encryption_spec_key_name set in aiplatform.init.
         """
         utils.validate_display_name(display_name)
 
         super().__init__(project=project, location=location, credentials=credentials)
         self._display_name = display_name
         self._project = project
+        self._training_pipeline_encryption_spec_key_name = (
+            training_pipeline_encryption_spec_key_name
+        )
+        self._model_encryption_spec_key_name = model_encryption_spec_key_name
         self._gca_resource = None
 
+    @property
+    def training_pipeline_encryption_spec(
```

Comment:
I'm not sure of the purpose of making this a public method. It doesn't seem like it would be useful for the user to access. If all this of this logic is centralized in the global config you can lock in this spec at construction time.

Reply:
Removed